compiler/rustc_typeck/src/check/intrinsic.rs

```diff
@@ -21,36 +21,68 @@ fn equate_intrinsic_type<'tcx>(
     tcx: TyCtxt<'tcx>,
     it: &hir::ForeignItem<'_>,
     n_tps: usize,
+    n_lts: usize,
     sig: ty::PolyFnSig<'tcx>,
 ) {
-    match it.kind {
-        hir::ForeignItemKind::Fn(..) => {}
+    let (gen_lts, gen_tys, gen_cns, span) = match &it.kind {
+        hir::ForeignItemKind::Fn(.., generics) => {
+            let mut gen_lts = 0;
+            let mut gen_tys = 0;
+            let mut gen_cns = 0;
```

Comment:
You could use `GenericParamCount` for this. See the function `own_counts`.

Reply:
The HIR and `tcx.generics_of(...).own_counts()` seem to disagree on the number of lifetime parameters sometimes, but this seems unrelated to my changes. Fixed.